Frameworks/Foundation/NSKeyValueCoding.mm

```diff
@@ -442,12 +442,17 @@ - (void)setValuesForKeysWithDictionary:(NSDictionary*)keyedValues {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSDictionary*)dictionaryWithValuesForKeys:(NSArray*)keys {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSDictionary* results = [NSMutableDictionary dictionary];
```

Comment:
You can give it a hint here, and say `dictionaryWithCapacity:keys.count` -- just to reduce the burden of re-hashing if keys is long :)